### PR TITLE
sys: random: add xorshift prng

### PR DESCRIPTION
**Pull request by kaspar030**
~~Waiting for #7544.~~

See https://en.wikipedia.org/wiki/Xorshift.

### CODE REVIEW

**Comment by kaspar030** [comment]
Dependency merged, ready for review.

File: sys/random/xorshift.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+
+#include "random.h"
+
+/* The state word must be initialized to non-zero */
+uint32_t xorshift32(uint32_t *state)
```
Comment (cgundogan) [inline sys/random/xorshift.c]: minor: why did you change the signature from the wikipedia example?
In the example it is:
```c
uint32_t state[static 1]
```

**Comment by Kijewski** [comment]
Please add more information: After how many values can an attacker devine the state of the system (the answer is 1 for the 32bit version, isn't it)? After how many invocations does the sequence repeat?

File: sys/random/xorshift.c (hunk 1 above)
Comment (kaspar030) [reply sys/random/xorshift.c]: I'd never seen that notation before, and as far as I can tell, using a pointer is equivalent, right?

Comment (cgundogan) [reply sys/random/xorshift.c]: but you kept the same notation at
https://github.com/RIOT-OS/RIOT/pull/7545/files#diff-82a7ba313c18943a85a93c8005a1795eR41

my google-fu tells me that compilers might do some optimizations when using this notation (I also did not see that notation before). However, I would prefer the pointer syntax.

**Comment by kaspar030** [comment]
> After how many values can an attacker devine the state of the system (the answer is 1 for the 32bit version, isn't it)?

I guess so, as there are only 2^32-1 possible seed states.

> After how many invocations does the sequence repeat?

Best guess: after about 2^32-1?

This should obviously not be used for crypto or anything sensitive.

File: sys/random/xorshift.c (hunk 1 above)
Comment (kaspar030) [reply sys/random/xorshift.c]: I'll change the other place.

Comment (kaspar030) [reply sys/random/xorshift.c]: done

**Comment by PeterKietzmann** [comment]
@kaspar030 thanks for squashing. Any objections using this implementation as default?

**Comment by kaspar030** [comment]
> ACK

merge at will...
